Calculate and Operate tests

alex1779 opened the pull request:
In this PR we:

- added calculate and operate tests
- fix linter errors
- add snapshop test for home component

alex1779 committed:
Calculate and Operate tests

alex1779 committed:
Fix linter errors

Estete9 committed:
Add snapshot test for Home component

shella12 reviewed (commented):
Hi 

#  Status: Invalidated :red_circle: 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that does not have descriptive PR description. Please add a short summary of changes made in this PR.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

roshan-bajgain commented on src/tests/opertate.test.js:
```diff
@@ -0,0 +1,9 @@
+import operate from '../logic/operate';
+
+const numberOne = 2;
+const numberTwo = 4;
+const operation = 'x';
+
+test('return the result of multiplication of two numbers', () => {
+  expect(operate(numberOne, numberTwo, operation)).toEqual('8');
+});
```
- [x] If you see the project requirement then you should test all the functionalities & scenarios of operate.js that is `+,-,*,/,%`. etc. Please write the test for all the functionality.

roshan-bajgain commented on src/tests/home.test.js:
```diff
@@ -0,0 +1,9 @@
+import renderer from 'react-test-renderer';
+import React from 'react';
+import Home from '../components/Home';
+
+test('Home component snapshot', () => {
+  const component = renderer.create(<Home />);
+  const tree = component.toJSON();
+  expect(tree).toMatchSnapshot();
+});
```
- [x] As the project requirement you should create unit tests for all the react components. So please create tests for all other components as well. You did not test all the component that is in the component folder.

roshan-bajgain commented on src/tests/calculate.test.js:
```diff
@@ -0,0 +1,15 @@
+import calculate from '../logic/calculate';
+
+test('clear display when button is clicked', () => {
+  const buttonName = 'AC';
+  const objNull = {
+    total: null,
+    next: null,
+    operation: null,
+  };
```
- [x] There are other functionality in the calculate.js file please make sure to test that functionality as well.
- [x] when clicking digits (0-9),  when clicking the decimal point ('.'), when clicking the '=' button to perform a calculation, please write all the test.

roshan-bajgain reviewed (changes requested):
Hi @alex1779 and @Estete9, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
- No linter errors. ✔️
- GitHub flow is followed. ✔️
- Some test are passing ✔️ 
## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

alex1779 committed:
Update test file

alex1779 committed:
test all the functionalities & scenarios of operate.js

Estete9 committed:
Implement test for Quotes.js layout is correct

Estete9 committed:
Implement Quotes test with mock response

Installed msw to create a mock response
Created a mock.js file with the mock response using msw, including the id
Implemented the test using msw/node to mock the response modeled with
mock.js
Used async and await to wait for the component to fully render and for
the mock response to finish

Estete9 committed:
Implement Quotes test with mock response

Installed msw to create a mock response
Created a mock.js file with the mock response using msw, including the id
Implemented the test using msw/node to mock the response modeled with
mock.js
Used async and await to wait for the component to fully render and for
the mock response to finish

Estete9 committed:
Merge branch 'test' of https://github.com/alex1779/math-magicians into test

Estete9 committed:
Merge branch 'test' of https://github.com/alex1779/math-magicians into test

remove console.log

Estete9 committed:
Merge branch 'test' of https://github.com/alex1779/math-magicians into test

Estete9 committed:
Implement test for Errorpage.js

Estete9 committed:
Implement Navbar.js test

Estete9 committed:
Implement Router.js test

Estete9 committed:
Implement tests for three components in buttons.js

VICTOR-CHUKWU reviewed (approved):
Hi @alex1779  
# STATUS: APPROVED! 🥇 💯 
![done](https://media.giphy.com/media/l1uga8ObPBnIBoXUQ/giphy.gif)

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- [ ] All test passed
- [ ] all methods tested and compoenets tested

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
